### DISCUSSION TITLE
DAQmx - Crash when attempting to read analog inputs - protected memory error

### DISCUSSION THREAD

**Discussion by JordLab** [bonsai-rx/bonsai#1638] [Q&A]
Hi All,

Very much enjoying setting up bonsai to gather fiber photometry data, behavioral videos, and analogue inputs!

However, whenever I attempt to use AnalogInput, Bonsai crashes. In the terminal it says 'UnhandledException: System.AccessViolationException: Attempted to read or write protected memory'. Other users with this issue seemed to fix it by downgrading DAQmx to previous versions, but I have now tried several different earlier versions, and no luck. Is there a way to know which version Bonsai will accept? Does anyone have any other ideas?

**Comment by bruno-f-cruz** [comment]
What versions have you tried? I would advise the 19.0

**Reply by JordLab** [reply]
I have tried 19.0 - actually, this fixes the issue on my laptop, but not the PC that we are using for data acquisition. It still has the same error there even with versions 19.0, and 21.8 - both versions I know worked for others.
